src/fetch.ts

```diff
@@ -22,6 +25,28 @@ export interface $Fetch {
   raw<T = any>(request: FetchRequest, opts?: FetchOptions): Promise<FetchResponse<T>>
 }
 
+export function setHeader (options: FetchOptions, _key: string, value: string) {
+  const key = _key.toLowerCase()
+  options.headers = options.headers || {}
+  if (Array.isArray(options.headers)) {
+    const existingHeader = options.headers.find(([header]) => header.toLowerCase() === key)
+    if (existingHeader) {
+      existingHeader[1] = value
+    } else {
+      options.headers.push([key, value])
+    }
+  } else if ('set' in options.headers) {
+    ;(options.headers as Headers).set(key, value)
+  } else {
+    const existingHeader = Object.keys(options.headers).find(header => header.toLowerCase() === key)
```

Comment:
By early returning on `set in` case, we can dedup this line

Reply:
not quite sure how to dedupe it - though it feels very close with the array/object syntax. how were you thinking?

Reply:
By early return in first if case we can move variable to outer scope :)